### PR TITLE
QE: Remove XEN test coverage

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

Card: https://github.com/SUSE/spacewalk/issues/19103

Our team agreed that XEN feature triggers almost the same lines of code than KVM does.
This test consumes resources and time for not that much benefit. So we decided to keep only KVM test coverage and remove XEN from our test environment (saving hardware resources, test suite time, and getting rid of some flaky tests due to timeouts giving false alarms).

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
- Cucumber tests were deleted

- [x] **DONE**

## Links

Ports:
- Manager-4.2
- Manager-4.3

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: testsuite/documentation/optional.md [hunk 1]
```diff
@@ -255,27 +254,24 @@ export VIRTHOST_KVM_PASSWORD=therootpwd
...
-A `/var/testsuite-data/disk-image-template-xenpv.qcow2` disk image to use for Xen ParaVirtualized
-guests is needed on the machine pointed by `VIRTHOST_XEN_URL`.
 
 In order for the virtual hosts to be able to report to the test server,
 use a bridge virtual network for the test machines.
 
 The `disk-image-template.qcow2` virtual disk image should
 have avahi daemon installed and running at first boot, and should be capable to be booted
-as either a Xen HVM or KVM guest. The disk images used by sumaform are good candidates
+as either a KVM guest. The disk images used by sumaform are good candidates
```
Comment (vandabarata) [inline testsuite/documentation/optional.md:263]:
```suggestion
as a KVM guest. The disk images used by sumaform are good candidates
```

File: testsuite/features/step_definitions/salt_steps.rb [hunk 2]
```diff
@@ -438,9 +438,8 @@ def pillar_get(key, minion)
 end
 
 When(/^I enter "([^"]*)" password$/) do |host|
-  raise "#{host} minion password is unknown" unless %w[kvm_server xen_server].include?(host)
+  raise "#{host} minion password is unknown" unless %w[kvm_server].include?(host)
```
Comment (Bischoff) [inline testsuite/features/step_definitions/salt_steps.rb:441]: simplify to:
```
unless host == 'kvm_server'
```

**Review by Bischoff — CHANGES_REQUESTED** [review]
just a small nitpick

**Review by nodeg — APPROVED** [review]
Other than Vandas and Erics suggestions it LGTM.

**Comment by srbarrios** [comment]
The Rubocop issue is not related to this PR, as agreed with QE Squad, merging as it is.

**Pull request merged**